src/PrizeVault.sol

```diff
@@ -0,0 +1,750 @@
+
+        uint256 _shares = previewDeposit(_assets);
+        _depositAndMint(_owner, _owner, _assets, _shares);
+
+        if (twabController.delegateOf(address(this), _owner) != SPONSORSHIP_ADDRESS) {
+            twabController.sponsor(_owner);
+        }
+
+        emit Sponsor(_owner, _assets, _shares);
+
+        return _shares;
+    }
+
+    /* ============ Yield Functions ============ */
+
+    /**
+     * @notice Total available yield on the vault.
+     * @return The available yield balance
+     */
+    function availableYieldBalance() public view returns (uint256) {
+        uint256 _totalAssets = totalAssets();
+        uint256 _allocatedAssets = totalSupply() + yieldBuffer;
+        if (_allocatedAssets >= _totalAssets) {
+            return 0;
+        } else {
+            unchecked {
+                return _totalAssets - _allocatedAssets;
+            }
+        }
+    }
+
+    /* ============ LiquidationSource Functions ============ */
+
+    /// @inheritdoc ILiquidationSource
+    /// @dev Returns the withdrawable amount of the available yield minus any yield fees.
+    function liquidatableBalanceOf(address _token) public view returns (uint256) {
+        if (_token != address(_asset)) {
+            return 0;
+        } else {
+            uint256 _availableYieldBalance = availableYieldBalance();
+            uint256 _maxWithdraw = yieldVault.maxWithdraw(address(this)) + _asset.balanceOf(address(this));
+            uint256 _withdrawableYieldBalance = _availableYieldBalance >= _maxWithdraw ? _maxWithdraw : _availableYieldBalance;
+            return _withdrawableYieldBalance.mulDiv(FEE_PRECISION - yieldFeePercentage, FEE_PRECISION);
+        }
+    }
+
+    /// @inheritdoc ILiquidationSource
+    function transferTokensOut(
+        address,
+        address _receiver,
+        address _tokenOut,
+        uint256 _amountOut
+    ) external onlyLiquidationPair returns (bytes memory) {
+        if (_tokenOut != address(_asset)) {
+            revert LiquidationTokenOutNotAsset(_tokenOut, address(_asset));
+        }
+
+        uint32 _yieldFeePercentage = yieldFeePercentage;
+        if (_amountOut == 0) revert LiquidationAmountOutZero();
+
+        uint256 _availableYieldBalance = availableYieldBalance();
+        address _yieldFeeRecipient = yieldFeeRecipient;
+
+        // Determine the proportional yield fee based on the amount being liquidated:
+        uint256 _yieldFee;
+        if (_yieldFeePercentage != 0 && _yieldFeeRecipient != address(0)) {
+            _yieldFee = (_amountOut * FEE_PRECISION) / (FEE_PRECISION - _yieldFeePercentage) - _amountOut;
+        }
+
+        // Ensure total withdrawn does not exceed the available yield balance:
+        uint256 _totalToWithdraw = _amountOut + _yieldFee;
+        if (_totalToWithdraw > _availableYieldBalance) {
+            revert LiquidationExceedsAvailable(_totalToWithdraw, _availableYieldBalance);
+        }
+
+        _withdraw(address(this), _totalToWithdraw);
```

Comment:
We should be minting vault shares to the fee recipient rather than withdrawing from the yield vault, as it allows them to withdraw at their leisure, allows us to liquidate debt, and likely uses less gas.